Started Configurations Revamp in new branch

angelac2023 opened the pull request:
We would really appreciate some feedback on our progress.

amgrubb committed:
Rename Models file.

amgrubb committed:
Simplified Analysis Inspector

angelac2023 committed:
Assignments button disappears after switching to Analysis mode

angelac2023 committed:
Moved the sidebar from the left to the right.

Manarnz committed:
Solved the analysisInspector clearInspector problem

Previously, everytime we click add new config (request) or we switch back and forth in model-analysis modes, a new analysisInspector would be appended. This is because we did not have the event 'clearInspector .inspector-views' : 'removeView'
Now, it is connected and updated to each config (request)

angelac2023 committed:
Created buttons in main toolbar

leshghi commented on leaf-analysis/app.js:
```diff
@@ -4,7 +4,7 @@
 // Name of .jar file for BloomingLeaf project must be Blooming.jar
 //var userPath = "/Users/<your user path here>/BloomingLeaf"
-var userPath = "/Users/judySmith/git/BloomingLeaf"
+var userPath = "/Users/ChuAn_nu0wj7q/Desktop/BloomingLeaf"
```
Please remove this file from PR

amgrubb replied:
Agreed this should be removed.

leshghi commented on leaf-ui/css/style.css:
```diff
@@ -189,14 +180,40 @@ body.waiting *
+  margin-top: 0px; 
+}
+
+#configID { 
```
Why do we have empty fields in the css?

leshghi replied:
^ @angelac2023 @Manarnz

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -115,6 +115,14 @@ $('#btn-view-intermediate').on('click', function() {
 });
 
+$('#simulate-single-path-btn').on('click', function() { 
+    console.log("Waiting to connect to backbone!"); 
```
I would remove the console logs and add a TODO to implement

leshghi replied:
^ @angelac2023 , @Manarnz

leshghi reviewed (commented):
This looks like a good start to me. The biggest issue for me currently is that when you select a config and then switch back the params resets back to the default values, and you need to click again to see the config's actual values. Please also remove the now unused functions in AnalysisInspector.js. There are also some CSS things I personally would change (big gap between analysis params title and input boxes, maybe a line between the analysis params area and the config sidebar, input boxes too narrow, etc) but those are minor things and can be dealt with later. I defer to @amgrubb's review on what they want to change/prioritize.

amgrubb reviewed (changes requested):
Please make these updates, so I can run it locally.

Manarnz committed:
removed changes from app.js

Manarnz committed:
Implemented Ali's comments

We deleted the functions in AnalysisInspector.js and fixed some minor css issues that Ali pointed out.

angelac2023 committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into maac-moveAnalysis

angelac2023 committed:
Solved bug in ConfigInspector and onFunctions.js

angelac2023 committed:
Removed app.js commit

angelac2023 committed:
Deleted app.js

angelac2023 committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into maac-moveAnalysis

angelac2023 committed:
Merge branch 'maac-moveAnalysis' of https://github.com/amgrubb/BloomingLeaf into maac-moveAnalysis

leshghi commented on leaf-analysis/app.js:
```diff
@@ -114,4 +114,4 @@ function passIntoJar(res) {
 
-server.listen(8080);
\ No newline at end of file
+server.listen(8080);
```
Remove this file from commit

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -3,6 +3,8 @@ This file contains all the jQuery functions that are associated with buttons and
 It also contains the setup for Rappid elements.
 */
 
+// const { config } = require("process");
```
Remove this

leshghi commented on leaf-ui/rappid-extensions/BackboneModelsAnalysis.js:
```diff
@@ -72,7 +72,8 @@ var ConfigModel = Backbone.Model.extend({
         selected: true,
-        results : new ResultCollection([])
+        results : new ResultCollection([]),
+        mode: true,
```
What does mode represent?

leshghi commented on leaf-ui/rappid-extensions/ConfigInspector.js:
```diff
@@ -101,13 +101,17 @@ var Config = Backbone.View.extend({
     initialize: function(){
         this.innerView = new ResultsDropdown({collection:this.model.get("results"), config: this.model});
         this.innerView.render();
+        // removes configurations
```
I don't know if these comments are necessary/accurate. I defer to @amgrubb on that though. If you leave them in the correct style is `// First word capitalized`

leshghi commented on leaf-ui/rappid-extensions/ConfigInspector.js:
```diff
@@ -123,21 +127,27 @@ var Config = Backbone.View.extend({
        '</script>'].join(''),
 
     events: {
+        //enables switching of results 
```
Same comment as above re: these comments

leshghi commented on leaf-ui/rappid-extensions/ConfigInspector.js:
```diff
@@ -316,20 +319,35 @@ var ConfigInspector = Backbone.View.extend({
     /** Add a new configuration view to the sidebar */
     loadConfig : function(config) {
         var view = new Config({model: config});
+        //view.showAnalysisInspector();
```
Remove

leshghi reviewed (changes requested):
I like the concept - for me the biggest bug is that now when you delete the first config the entire sidebar disappears when you switch to modelling mode and then back to analysis mode. Here is a thought - you already have a selected parameter - what if you just used that instead of mode to make sure the previously selected configuration is the one that has its analysisInspector rendered?

amgrubb committed:
Renamed BBM elements.

angelac2023 committed:
Implemented Ali's comments in PR

angelac2023 committed:
Added files to commit

angelac2023 committed:
Implemented Ali's comments from pull request

angelac2023 committed:
Revert "Implemented Ali's comments from pull request"

This reverts commit b1a837d6cf35ac7f3ada1ba7ba630a29656e2f8f, reversing
changes made to 142b9940eeb8dab2d77a3ab33aee7b978ca26129.

angelac2023 committed:
Revert "Revert "Implemented Ali's comments from pull request""

This reverts commit d4f70669731466e1791101828cca5347267df523.

angelac2023 committed:
Added lab bee favicon

angelac2023 committed:
Merge branch 'maac-moveAnalysis' of https://github.com/amgrubb/BloomingLeaf into maac-moveAnalysis

angelac2023 committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into maac-moveAnalysis

angelac2023 committed:
Added image of bee to project

Manarnz committed:
renamed lab_bee to lab-bee

Manarnz committed:
Revert "renamed lab_bee to lab-bee"

This reverts commit 3b9c2ad88713497f3c9ba6904947bb6806f58cae.

Manarnz committed:
Revert "Added image of bee to project"

This reverts commit 26cfe2a9dce379f80efaee549bcb43bdab3c0cbf.

Manarnz committed:
Revert "Added lab bee favicon"

This reverts commit b1837354b2a71d616dc3fa4ff073da92bc015267.

leshghi commented on leaf-analysis/app.js:
```diff
@@ -113,5 +113,3 @@ function passIntoJar(res) {
 }
 
```
Can you please remove this file from your commit, thanks

leshghi replied:
Also deleting this line makes it impossible to actually run (and therefore test) the web page.

leshghi commented on leaf-ui/css/style.css:
```diff
@@ -140,16 +140,7 @@ body.waiting *
-  display: block;
-}
-
-#analysis-sidebar a:hover {
```
Why did you remove the hovering feature and would it be possible to bring it back?

angelac2023 replied:
We initially removed it because it was making the configurations sidebar look rather odd. We will add it back in the next commit.

leshghi commented on leaf-ui/index.html:
```diff
@@ -132,6 +137,7 @@
             <div id="layout"></div>
         </div>
 
+        
```
Remove this line

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -411,39 +418,46 @@ function reassignIntentionIDs() {
     configInspector.render();
-	$('#stencil').css("display", "none");
+
+    // disappear
```
Remember my previous review where I mentioned correct comment format is // First word capitalized

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -411,39 +418,46 @@ function reassignIntentionIDs() {
+	$('.link-tools .tool-remove').css("display", "none");
+    $('.link-tools .tool-options').css("display", "none");
+    
+    // appear 
```
Same thing about commenting convention

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -411,39 +418,46 @@ function reassignIntentionIDs() {
-    $('.model-clears').css("display", "none");
     $('.analysis-clears').css("display", "");
+    
+    //$('#on-off').css("display", "none");
```
Why did you move this commented out code?

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -458,11 +472,11 @@ $('#model-cur-btn').on('click', function() {
 
 	// Cleaning the previous analysis data for new execution
 	//globalAnalysisResult.elementList = "";
+
```
No need to touch these lines

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -458,11 +472,11 @@ $('#model-cur-btn').on('click', function() {
     savedAnalysisData.finalValueTimePoints="";
     
     analysisRequest.action = null;
-
```
No need to touch this line

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -526,21 +541,28 @@ function switchToModellingMode() {
+    $('#on-off').css("display", "");
 
-    $('#stencil').css("display","");
+    // disappear 
```
Same thing about commenting convention

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -526,21 +541,28 @@ function switchToModellingMode() {
     // previousModel is NOT of type Model
     previousModel = JSON.parse(JSON.stringify(model));
+    
+    $('#on-off').css("display", "");
```
If we are re-sectioning these functions into appear and disappear, why is this in a separate section?

leshghi reviewed (commented):
Once you respond to the comments and make the changes I will approve. Also, #465 should follow this PR if you are looking for something to work on once it is merged.

angelac2023 committed:
Fixed bug where you are unable to switch back to modelling mode after adding an element

Manarnz committed:
Implemented Ali's comments

Manarnz committed:
Clean up the code

leshghi reviewed (approved):
Thanks for making these changes

leshghi commented on leaf-ui/js/onFunctions.js:
```diff
@@ -874,6 +873,7 @@ paper.on({
                     actorInspector.render();
                 } else {
                     var elementInspector = new ElementInspector({model: cell});
+                    console.log(elementInspector); 
```
Remove this line

amgrubb commented on leaf-ui/rappid-extensions/AnalysisInspector.js:
```diff
@@ -85,102 +79,4 @@ var AnalysisInspector = Backbone.View.extend({
 		}
 	},
-
-	/**
```
Note: This code should not have been deleted.

leshghi committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into maac-moveAnalysis

leshghi committed:
Quick post-merge cleanup
